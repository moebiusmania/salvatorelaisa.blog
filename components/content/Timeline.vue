<script setup lang="ts">
interface Props {
  events: string;
  years: string
}

const { events = "", years = "" } = defineProps<Props>();

const getItems = (list: string): string[] => list.split(",").map((item) => item.trim());

// const isEven = (n: number): boolean => (n % 2) === 0
</script>

<template>
  <section class="not-prose">
    <ul class="timeline timeline-vertical timeline-compact">
      <li v-for="(event, index) in getItems(events)" :key="index">
        <hr v-if="index !== 0" />
        <div class="timeline-start">{{ getItems(years)[index] }}</div>
        <div class="timeline-middle text-primary">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor">
            <path fill-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.857-9.809a.75.75 0 00-1.214-.882l-3.483 4.79-1.88-1.88a.75.75 0 10-1.06 1.061l2.5 2.5a.75.75 0 001.137-.089l4-5.5z"
              clip-rule="evenodd" />
          </svg>
        </div>
        <div class="timeline-end timeline-box">{{ event }}</div>
        <hr v-if="index !== (getItems(events).length - 1)" />
      </li>
    </ul>
  </section>
</template>

<style scoped>
section {
  & ul {
    & div.timeline-start {
      font-weight: 700;
    }

    & svg {
      width: 1.25rem;
      height: 1.25rem;
    }
  }
}
</style>