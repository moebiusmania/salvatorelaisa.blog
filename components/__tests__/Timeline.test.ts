import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Timeline from '../../components/content/Timeline.server.vue'

describe('Timeline', () => {
  it('renders correctly with provided items', () => {
    const items = 'Item 1,Item 2,Item 3'
    const wrapper = mount(Timeline, {
      props: {
        items
      }
    })

    // Check if all items are rendered
    const listItems = wrapper.findAll('li')
    expect(listItems).toHaveLength(3)
    expect(listItems[0].text()).toBe('Item 1')
    expect(listItems[1].text()).toBe('Item 2')
    expect(listItems[2].text()).toBe('Item 3')
  })

  it('renders empty list when no items provided', () => {
    const wrapper = mount(Timeline, {
      props: {
        items: ''
      }
    })

    const listItems = wrapper.findAll('li')
    expect(listItems).toHaveLength(1)
    expect(listItems[0].text()).toBe('')
  })

  it('matches snapshot', () => {
    const items = 'Event 1,Event 2,Event 3'
    const wrapper = mount(Timeline, {
      props: {
        items
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})