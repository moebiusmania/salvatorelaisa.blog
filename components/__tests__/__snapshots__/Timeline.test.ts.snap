// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Timeline > matches snapshot 1`] = `
"<ul data-v-8c6e1942="" class="timeline">
  <li data-v-8c6e1942="">Event 1</li>
  <li data-v-8c6e1942="">Event 2</li>
  <li data-v-8c6e1942="">Event 3</li>
</ul>"
`;
